docs/build-macos.md

```diff
@@ -0,0 +1,52 @@
+macOS builds can be created using:
+
+- Clang, using Meson buildsystem and Homebrew for installing dependencies
+  (*recommended*)
+- GCC, using Meson buildsystem and Homebrew for installing dependencies.
+- Clang or GCC, using Meson buildsystem and MacPorts for dependencies.
```

Comment:
(direct address) macOS builds can be created using Meson, compiled using Clang or GCC, and provided with dependencies using Brew or MacPorts.

(with adjective) macOS builds can be created using the Meson buildsystem, compiled using the Clang or GCC compilers, and provided with dependencies using the Brew or MacPorts package managers.

(adjective on meson only, because it's lesser-known) macOS builds can be created using the Meson buildsystem, compiled using Clang or GCC, and provided with dependencies using Brew or MacPorts.